lightwood/analysis/nn_conf/temp_scale.py

```diff
@@ -0,0 +1,100 @@
+from copy import deepcopy
+from typing import Dict, Tuple
+from types import SimpleNamespace
+
+import torch
+import pandas as pd
+from torch import nn, optim
+from sklearn.preprocessing import OrdinalEncoder
+
+from lightwood.helpers.log import log
+from lightwood.analysis.base import BaseAnalysisBlock
+
+
+class TempScaler(BaseAnalysisBlock):
```

Comment:
I honestly spent like 20 mins pouring over this and I don't quite understand why it should work better than just taking the probability and calling it a confidence... but I'm tired, if you think it's worth it feel free to merge :) but see 2 comments below first.

Reply:
Here's a good reference (sections 4.2 at the end, then results in table 1): https://arxiv.org/pdf/1706.04599.pdf

TL;DR: the raw output of a softmax in a classifier is rarely well-calibrated. Turns out learning a value for the operation's "temperature" parameter can help quite a bit in terms of calibration.